packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -77,6 +117,8 @@ export class Outbox {
 	}
 
 	public submitAttach(message: BatchMessage) {
+		this.maybeFlushPartialBatch(this.mainBatch, message);
```

Comment:
Shouldn't this apply to attach message too? What I mean is attach messages batch should also have the same reference sequence number right?

Reply:
It does not matter, but I think we want to get to the point where this extra arg is required, so yes, we should add it to all calls.

Reply:
Ohhhh... Nice catch.

Reply:
@vladsud Why do you say it does not matter? How is attach messages batch different from regular batch?

Reply:
Yeah, this is a legitimate bug. The current code will flush the main batch (only) when it detects an attach op with a different reference sequence number. I think the `maybeFlushPartialBatch` should flush both batches when this happens for either attach ops or regular ops, as the reference sequence number mismatch will be in both.

Adding a unit test for these cases right now.

Reply:
updated @agarwal-navin @vladsud

Reply:
What I meant - ref seq number of attach op does not matter, as result of applying it does not matter when client send it (what state it had when sending it)

Reply:
BTW, we should reconsider having separate batch for attach ops. It was more of a short term solution to reduce batch sizes. With compression and chunking the value of this feature is lower. It gives us maybe some benefit on latency (ops go out sooner and applied sooner), but it also gives us drawback in terms of lower compression rates and more (overall) messages being sent (that will mean more ops once we have seq number virtualization). I'm not sure benefits outweigh cons (including more code / more complexity), and we do not have good way to measure  impact here, so I think removing all this logic is fine.

Reply:
I like the idea of removing the separate batch for attach ops. It adds additions flows for sending batches which increases complexity and thereby changes of things going wrong (this issue that was luckily caught and fixed for example).

Reply:
Yeah, we will do that after compression/chunking goes mainstream (opt-out, not opt-in)